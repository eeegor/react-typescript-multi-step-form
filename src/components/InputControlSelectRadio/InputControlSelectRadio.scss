.input-control-select-radio__inputs {
	display: flex;
	flex-wrap: wrap;
	justify-content: center;
}
