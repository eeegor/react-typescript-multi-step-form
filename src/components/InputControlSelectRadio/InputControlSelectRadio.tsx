/**
 * Component InputControlSelectRadio
 */

import * as React from 'react';
import * as classnames from 'classnames';
import { InputControlRadio, InputControl, Label, Info } from '..';
import './InputControlSelectRadio.scss';

export interface InputControlSelectRadioProps {
	id: string;
	name: string;
	label?: string;
	values: string[];
	info?: string;
	value?: string;
	status?: 'success' | 'danger';
	required?: boolean;
	className?: boolean;
	defaultChecked?: string;
	onChange?: (name: string, event: React.ChangeEvent<HTMLInputElement>) => void;
}

export class InputControlSelectRadio extends React.Component<InputControlSelectRadioProps> {
	render(): JSX.Element {
		const {
			id,
			className,
			required,
			onChange,
			label,
			name,
			defaultChecked,
			info,
			status,
			values,
		} = this.props;

		return (
			<InputControl
				id={id}
				type={'select-radio'}
				status={status}
				className={classnames(className)}
			>
				<Label className="input-control__label" htmlFor={id || name} label={label || ''} />
				<Info className="input-control__info" label={info || ''} />
				<div className="input-control-select-radio__inputs">
					{values.map((item: string, index: number) => (
						<InputControlRadio
							id={`${id}-${index + 1}`}
							required={required}
							key={index}
							name={name}
							label={item}
							value={item || ''}
							defaultChecked={defaultChecked === item}
							onChange={(radio, event) => onChange && onChange(radio, event)}
						/>
					))}
				</div>
			</InputControl>
		);
	}
}
