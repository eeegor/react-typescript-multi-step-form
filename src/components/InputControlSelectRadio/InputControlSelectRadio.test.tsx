import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { InputControlSelectRadio } from './InputControlSelectRadio';

describe('InputControlSelectRadio', () => {
	const callback = jest.fn();
	const testData = {
		id: 'name',
		name: 'name',
		label: 'Your Name',
		values: ['one', 'two'],
		onChange: callback,
	};

	it('renders markup correctly', () => {
		const component = <InputControlSelectRadio {...testData} />;
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should adopt a valid status attribute', () => {
		const status = 'danger';
		const wrapper = mount(<InputControlSelectRadio status={status} {...testData} />);
		expect(
			wrapper
				.find('.input-control')
				.first()
				.hasClass('input-control--danger')
		).toEqual(true);
	});

	it('should adopt a valid info attribute', () => {
		const info = 'test';
		const wrapper = mount(<InputControlSelectRadio info={info} {...testData} />);
		expect(
			wrapper
				.find('.input-control__info')
				.first()
				.text()
		).toEqual('test');
	});

	it('change event works', () => {
		const wrapper = mount(<InputControlSelectRadio {...testData} />);
		wrapper
			.find('input[type="radio"]')
			.first()
			.simulate('change', { target: { checked: true } });
		expect(callback).toHaveBeenCalledTimes(1);
	});
});
