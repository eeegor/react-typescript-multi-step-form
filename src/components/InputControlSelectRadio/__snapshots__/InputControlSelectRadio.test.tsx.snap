// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControlSelectRadio renders markup correctly 1`] = `
<div
  className="input-control input-control-select-radio"
  id="input-control-name"
>
  <label
    className="label input-control__label"
    htmlFor="name"
  >
    Your Name
  </label>
  <div
    className="input-control-select-radio__inputs"
  >
    <div
      className="input-control input-control-radio"
      id="input-control-name-1"
    >
      <input
        autoFocus={false}
        className="input-control__input input-radio"
        defaultChecked={false}
        id="name-1"
        name="name"
        onChange={[Function]}
        type="radio"
        value="one"
      />
      <label
        className="label input-control__label"
        htmlFor="name-1"
      >
        one
      </label>
    </div>
    <div
      className="input-control input-control-radio"
      id="input-control-name-2"
    >
      <input
        autoFocus={false}
        className="input-control__input input-radio"
        defaultChecked={false}
        id="name-2"
        name="name"
        onChange={[Function]}
        type="radio"
        value="two"
      />
      <label
        className="label input-control__label"
        htmlFor="name-2"
      >
        two
      </label>
    </div>
  </div>
</div>
`;
